### PR TITLE
chore: neotraverse

### PR DESCRIPTION
**Pull request by PuruVJ**
## Description

Supersedes #1498. This makes a few change:
- tsconfig: Add moduleResolution and module: NodeNext, needed for subpath import(neotraverse/legacy)
- Add svelte/types/compiler/interfaces to `paths` field, required because moduleResolution is strict, and doesn't allow importing from any arbitrary file, hence this workaround required.
- Bump `astring`: `astring` fixed export types error in 1.8.5

Make sure to follow the PR preparation steps in [CONTRIBUTING.md](../CONTRIBUTING.md#preparing-your-pr) before submitting your PR:

- [x] format the codebase: from the root, run `yarn fmt:prettier`.
- [x] update all snapshots (in core & CLI): from the root, run `yarn test:update`
- [x] add Changeset entry: from the root, run `yarn g:changeset` and follow the CLI instructions. Alternatively, use the Changeset Github Bot to create the file.

### CODE REVIEW

File: packages/core/tsconfig.json [hunk 1]
```diff
@@ -14,7 +16,9 @@
     "baseUrl": ".",
     "paths": {
       "@builder.io/mitosis": ["./src"],
-      "@/*": ["./src/*"]
+      "@/*": ["./src/*"],
+      // To avoid error from moduleResolution: NodeNext
+      "svelte/types/compiler/interfaces": ["./node_modules/svelte/types/compiler/interfaces"]
```
Comment (samijaber) [inline packages/core/tsconfig.json:21]: lol that was probably a fun one

**Review by samijaber — APPROVED** [review]
Thanks @PuruVJ !! and @max-arias for the initial effort

File: .changeset/seven-drinks-bake.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+'@builder.io/mitosis': patch
+---
+
+Replaces the traverse dep for neotraverse to avoid unnecessary deps
```
Comment (samijaber) [inline .changeset/seven-drinks-bake.md:5]:
```suggestion
Replaces `traverse` dependency with the smaller `neotraverse`
```

File: packages/core/tsconfig.json (hunk 1 above)
Comment (PuruVJ) [reply packages/core/tsconfig.json:21]: ChatGPT had the fun 😆

**Comment by PuruVJ** [comment]
Tests are passing 🎉🎉🎉

**Comment by PuruVJ** [comment]
Thanks a lot @samijaber!! Huge thanks to @max-arias for initial work

**Comment by PuruVJ** [comment]
@samijaber it seems that publish flow failed due to yarn

**Comment by PuruVJ** [comment]
You might be interested 👀👀👀👀 https://x.com/puruvjdev/status/1813262143384658279

**Pull request merged**